Branch-SDK/src/main/java/io/branch/referral/Branch.java

```diff
@@ -2263,194 +2231,6 @@ private boolean checkIntentForUnusedBranchLink(Intent intent) {
-                                        String new_Identity_Id = respJson.getString(Defines.Jsonkey.IdentityID.getKey());
-                                        if (!prefHelper_.getIdentityID().equals(new_Identity_Id)) {
-                                            //On setting a new identity Id clear the link cache
-                                            linkCache_.clear();
-                                            prefHelper_.setIdentityID(new_Identity_Id);
-                                            updateRequestsInQueue = true;
-                                        }
-                                    }
-                                    if (respJson.has(Defines.Jsonkey.DeviceFingerprintID.getKey())) {
-                                        prefHelper_.setDeviceFingerPrintID(respJson.getString(Defines.Jsonkey.DeviceFingerprintID.getKey()));
-                                        updateRequestsInQueue = true;
-                                    }
-                                }
-
-                                if (updateRequestsInQueue) {
-                                    updateAllRequestsInQueue();
-                                }
-
-                                if (thisReq_ instanceof ServerRequestInitSession) {
-                                    setInitState(SESSION_STATE.INITIALISED);
-                                    thisReq_.onRequestSucceeded(serverResponse, branchReferral_);
-                                    if (!((ServerRequestInitSession) thisReq_).handleBranchViewIfAvailable((serverResponse))) {
-                                        checkForAutoDeepLinkConfiguration();
-                                    }
-                                    // Count down the latch holding getLatestReferringParamsSync
-                                    if (getLatestReferringParamsLatch != null) {
-                                        getLatestReferringParamsLatch.countDown();
-                                    }
-                                    // Count down the latch holding getFirstReferringParamsSync
-                                    if (getFirstReferringParamsLatch != null) {
-                                        getFirstReferringParamsLatch.countDown();
-                                    }
-                                } else {
-                                    // For setting identity just call only request succeeded
-                                    thisReq_.onRequestSucceeded(serverResponse, branchReferral_);
-                                }
-                            }
-                        } else {
-                            //Publish success to listeners
-                            thisReq_.onRequestSucceeded(serverResponse, branchReferral_);
-                        }
-                    }
-                }
-                networkCount_ = 0;
-                if (hasNetwork_ && initState_ != SESSION_STATE.UNINITIALISED) {
```

Comment:
I removed both conditions here because I don't either one makes sense, e.g. if SESSION_STATE.UNINITIALISED, then we should execute the request and return an error indicating that SESSION_STATE.UNINITIALISED.

wrt `hasNetwork_` I left a few comments on that already, the reasoning is the same though, shouldn't execute requests right away and let the client know why they are failing rather than holding back on executing those requests.